app/admin/SettingsProvider.scala

```diff
@@ -0,0 +1,76 @@
+package admin
+
+import cats.syntax.either._
+import com.amazonaws.services.s3.AmazonS3
+import com.typesafe.config.Config
+import monitoring.SafeLogger
+import play.api.mvc.Result
+
+abstract class SettingsProvider {
+
+  // Models the possibility of settings changing over the application life cycle.
+  // For example, if settings are read from S3, the file can be polled for changes,
+  // so that the setting updates can propagated without having to re-deploy the app.
+  def settings(): Settings
+}
+
+class LocalFileSettingsProvider private (initialSettings: Settings) extends SettingsProvider {
+  override def settings(): Settings = initialSettings
+}
+
+object LocalFileSettingsProvider {
+
+  def fromLocalFile(localFile: SettingsSource.LocalFile): Either[Throwable, SettingsProvider] =
+    Settings.fromLocalFile(localFile).map(new LocalFileSettingsProvider(_))
+}
+
+class S3SettingsProvider private (initialSettings: Settings) extends SettingsProvider {
+
+  // TODO: in a subsequent PR, poll S3 for changes.
+  override def settings(): Settings = initialSettings
+}
+
+object S3SettingsProvider {
+
+  def fromS3(s3: SettingsSource.S3)(implicit s3Client: AmazonS3): Either[Throwable, SettingsProvider] =
+    Settings.fromS3(s3).map(new S3SettingsProvider(_))
+}
+
+object SettingsProvider {
+
+  private def fromSettingsSource(source: SettingsSource)(implicit client: AmazonS3): Either[Throwable, SettingsProvider] = {
+    SafeLogger.info(s"loading settings from $source")
+    source match {
+      case s3: SettingsSource.S3 => S3SettingsProvider.fromS3(s3)
+      case localFile: SettingsSource.LocalFile => LocalFileSettingsProvider.fromLocalFile(localFile)
+    }
+  }
+
+  private def fromConfiguration(config: Config)(implicit s3Client: AmazonS3): Either[Throwable, SettingsProvider] =
+    SettingsSource.fromConfig(config).flatMap(fromSettingsSource)
+
+  def fromConfigurationUnsafe(config: Config)(implicit s3: AmazonS3): SettingsProvider =
```

Comment:
let's add, here or at the call site, a comment of the sort I conspicuously failed to provide myself - just spelling out why we're letting this exception escape

Reply:
Going to defer this to next PR if that's alright (?) as these utility methods are going to change.